src/NextGen/DonationForm/DataTransferObjects/DonationFormGoalData.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Give\NextGen\DonationForm\DataTransferObjects;
+
+use Give\Framework\Support\Contracts\Arrayable;
+use Give\Framework\Support\ValueObjects\Money;
+use Give\NextGen\DonationForm\Repositories\DonationFormRepository;
+use Give\NextGen\DonationForm\ValueObjects\GoalTypeOptions;
+
+/**
+ * @unreleased
+ */
+class DonationFormGoalData implements Arrayable
+{
+      /**
+     * @var int
+     */
+    public $formId;
+    /**
+     * @var array
+     */
+    public $formSettings;
+    /**
+     * @var false
+     */
+    public $isEnabled;
+    /**
+     * @var GoalTypeOptions
+     */
+    public $goalType;
+    /**
+     * @var int
+     */
+    public $targetAmount;
+
+    /**
+     * @unreleased
+     */
+    public function __construct(int $formId, array $formSettings)
+    {
+        $this->formId = $formId;
+        $this->formSettings = $formSettings;
+        $this->isEnabled = $formSettings['enableDonationGoal'] ?? false;
+        $this->goalType = new GoalTypeOptions($formSettings['goalType'] ?? GoalTypeOptions::AMOUNT);
+        $this->targetAmount = $this->formSettings['goalAmount'] ?? 0;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function getCurrentAmount(): int
+    {
+        /** @var DonationFormRepository $donationFormRepository */
+        $donationFormRepository = give(DonationFormRepository::class);
+
+        if ($this->goalType->isDonors()) {
+            return $donationFormRepository->getTotalNumberOfDonors($this->formId);
+        }
+
+        if ($this->goalType->isDonations()) {
+            return $donationFormRepository->getTotalNumberOfDonations($this->formId);
+        }
+
+        return $donationFormRepository->getTotalRevenue($this->formId);
+    }
+
+    /**
+     * @unreleased
+     */
+    public function toArray(): array
+    {
+        $currentAmount = $this->getCurrentAmount();
+
+        return [
+            'type' => $this->goalType->getValue(),
+            'enabled' => $this->isEnabled,
+            'show' => $this->isEnabled,
+            'currentAmount' => $currentAmount,
+            'currentAmountFormatted' => $this->goalType->isAmount() ? Money::fromDecimal(
+                $currentAmount,
+                give_get_currency()
+            )->formatToLocale() : $currentAmount,
```

Comment:
First, you shouldn't use this function without passing the locale. Otherwise, the locale is going to come from the server's locale settings, not the browser.

I saw your comment regarding decimals. Full disclosure, this took me a down a whole rabbit hole as I got curious. This is, no joke, the best way I could find to do it in PHP:
```php
$amount = 12300.45;
$formattedAmount = $fmt->formatCurrency($amount, 'USD');
$pattern = '/(.+)\\' . $decialSymbol . '\d+(.*)/';
$integerAmount = preg_replace($pattern, '${1}${2}', $formattedAmount);
```
That should handle whatever the decimal symbol is as well as not message up left or right-side currency symbols. There just doesn't seem to be a way to tell MoneyPHP or the Intl library to not include the decimal. So interesting. We could add a `$withoutDecimal` parameter to methods within the `CurrencyFacade` class if we felt this was useful. 🤷‍♂️ 

The JS Intl lib has a [formatToParts](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/formatToParts) method which makes this a lot easier on that side.

In short, we may simply be worth passing the amount as an integer along with the currency, and have the front-end format it. That way we get the benefits of the browser locale, and can more easily hide decimal amounts if we want to.

Reply:
Yeah another weird one.  I originally had all the amount formatting taking place on the front-end using `Intl` which might still end up making the most sense.  That's why I added that `useCurrencyFormatter` in the first place.

Reply:
Reformatted to let front-end handle formatting amounts: 577688ae